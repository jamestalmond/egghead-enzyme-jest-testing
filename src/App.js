import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () => <div>Testing</div>;
const Title = ({ text }) => <div>{text}</div>;

class App extends Component {
	render() {
		return (
			<div className="App">
				<header className="App-header">
					<img src={logo} className="App-logo" alt="logo" />
					<Title text="Foo" />
					<p>
						Edit <code>src/App.js</code> and save to reload.
					</p>
					<Test />
					<ul className="App-list">
						<li>Test 1</li>
						<li>Test 2</li>
						<li>Test 3</li>
					</ul>
					<a
						className="App-link"
						href="https://reactjs.org"
						target="_blank"
						rel="noopener noreferrer"
					>
						Learn React
					</a>
				</header>
			</div>
		);
	}
}

export default App;
