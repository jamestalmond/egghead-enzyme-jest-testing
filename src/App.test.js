import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<App />', () => {
	it('should render 1 p tag', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('p').length).toBe(1);
	});

	it('should render an element with the class .App-header', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('.App-header').exists()).toBe(true);
	});

	it('should render an ul tag with 3 children', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('ul').children().length).toBe(3);
	});

	it('should render an ul tag with the className App-list', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('ul').hasClass('App-list')).toBe(true);
	});

	it('should render an a tag with the text Learn React', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('a').text()).toBe('Learn React');
	});

	it('should find an element with the text prop', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('[text="Foo"]'));
	});

	it('should find an element with the specified alt text', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find({ alt: 'logo' }));
	});
});
