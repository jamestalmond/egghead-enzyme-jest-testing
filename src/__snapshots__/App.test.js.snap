// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <Title
      text="Foo"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <Test />
    <ul
      className="App-list"
    >
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
      <li>
        Test 3
      </li>
    </ul>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
